main.c

```diff
@@ -969,6 +978,38 @@ int __attribute__((used)) main(void) {
     alarm_reset();
     #endif
 
+    // Pystack variables have to exist even in stackless for the function calls
+    supervisor_allocation *pystack;
+    mp_int_t pystack_size = CIRCUITPY_PYSTACK_SIZE; // Use build default for now.
+    #if MICROPY_ENABLE_PYSTACK
+    // Allocate default at least temporarily, needed for getenv
+    pystack = allocate_memory(CIRCUITPY_PYSTACK_SIZE, false, false);
+
+    // Fetch value if exists from settings.toml
+    #if CIRCUITPY_OS_GETENV
+    // Init needed.
+    supervisor_allocation *heap = allocate_remaining_memory();
```

Comment:
This is extra and done at a deeper level. You'll want to do the pystack deeper as well so that its size can be changed between reloads.

Reply:
Now all are done right before heap allocations and are free'ed on vm cleanup.